### DISCUSSION TITLE
Can someone explain how to bypass the error "cannot sign an IPA file on non-Mac platform AIR sdk" ?

### DISCUSSION THREAD

**Discussion by stephanberger4** [airsdk/Adobe-Runtime-Support#2945] [Q&A]
Hi, 

First time that I see this error in 8 years.. When I click on "publish" in Adobe Animate with the last AIR SDK Version I've got this error : "cannot sign an IPA file on non-Mac platform AIR sdk". 
I understood that I have to install Java 16 (here https://www.oracle.com/java/technologies/javase/jdk16-archive-downloads.html) . 
I did. 
Restarted Animate. 
Nothing changed, still the same error 

I did the uncomment `DebugOut=true` in the adt.cfg file and saw the details in the adt.log file. 
I saw that it was, indeed, still using the 17 version of java instead of the 16 that I have installed. (`Cannot create sun security class: JRE version = 17.0.6+9-LTS-190, Java Home = C:\Program Files\Adobe\Adobe Animate 2024\jre`).

How/where can I tell it to use java 16 instead of 17 ?
Is it the solution or is there a new one for this problem ? 

Thx

**Comment by ajwfrost** [comment]
My assumption is that you'd need to rename the existing JRE folder (within Animate) to something as a back-up:
```
cd "C:\Program Files\Adobe\Adobe Animate 2024"
rename jre jre17
```
and then copy the JDK16 version into a new jre folder there... so that Animate just uses this (you'll need to have shut down Animate before making this change!)

thanks

**Comment by stephanberger4** [comment]
I've tried this... weirdly it's still using Java 17...
I've shut down Animate before and after

Envoyé à partir de Outlook pour Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by stephanberger4** [comment]
**fixed !** (partly) 
So I installed JAVA 11 (because I've read that some people fix this bug with Java 11). 
I replaced everything in the JRE folder with Java 11
https://streamable.com/5ax9p2
Then I launched Animate and hit the "publish" button of my AIR project. 

Animate throws an error

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/6382334/5058f3fc-1004-4e5d-96a9-4f15b373380e)

But the IPA is, after that, succefully created.
